src/main.ts

```diff
@@ -0,0 +1,8 @@
+export async function run(): Promise<void> {
+    if (process.argv.includes('--version')) {
+        console.log(require('./package.json').version);
```

Comment:
`package.json` is not located in the current directory, which I believe is the reason why you marked this PR as a draft. I suggest modifying this line as follows:
```
const packageJsonPath = path.resolve(__dirname, '../package.json');
const pkg = JSON.parse(fs.readFileSync(packageJsonPath, 'utf-8'));
console.log(pkg.version);
```